llvm/lib/Target/DirectX/DXILOpBuilder.h

```diff
@@ -42,6 +48,7 @@ class DXILOpBuilder {
 private:
   Module &M;
   IRBuilderBase &B;
+  Triple TargetTriple;
```

Comment:
It will simplify things to have the DXIL version and shader stage as members instead of the triple, and then we can just assume that they're valid like so:
```c++
DXILOpBuilder::DXILOpBuilder(Module &M, IRBuilderBase &B) : M(M), B(B) {
  Triple TT(M.getTargetTriple());
  DXILVersion = TT.getDXILVersion();
  assert(TT.isShaderStageEnvironment() &&
         "Cannot build DXIL ops outside of shader environment");
  ShaderStage = TT.getEnvironment();
}
```
It shouldn't be DXILOpBuilder's job to validate the triple.

Reply:
> It will simplify things to have the DXIL version and shader stage as members instead of the triple, and then we can just assume that they're valid like so:
> 
> ```c++
> DXILOpBuilder::DXILOpBuilder(Module &M, IRBuilderBase &B) : M(M), B(B) {
>   Triple TT(M.getTargetTriple());
>   DXILVersion = TT.getDXILVersion();
>   assert(TT.isShaderStageEnvironment() &&
>          "Cannot build DXIL ops outside of shader environment");
>   ShaderStage = TT.getEnvironment();
> }
> ```
> 
> It shouldn't be DXILOpBuilder's job to validate the triple.

Changed.